Say/ViewController.swift

```diff
@@ -128,5 +134,22 @@ class ViewController: NSViewController {
         catch {/* error handling here */}
     }
     
+    @IBAction func setAlarm(_ sender: NSControl) {
+        if alarmButton.state == NSOnState {
+            self.alarmTime = datePicker.dateValue
+            self.timer = Timer(fireAt: alarmTime, interval: 0, target: self, selector: #selector(doAlarm), userInfo: nil, repeats: false)
+            RunLoop.main.add(timer, forMode: RunLoopMode.commonModes)
+        } else if alarmButton.state == NSOffState {
+            self.timer.invalidate()
+        }
+    }
+    
+    func doAlarm() {
+        alarmButton.isEnabled = false
```

Comment:
Yes I think so. Because if this button is not be disabled there is an issue that button could be clicked during speech. This causes program to says text again and again. Is there any good idea to fix this issue?